src/Perlang.Interpreter/NameResolution/NameResolver.cs

```diff
@@ -425,7 +425,7 @@ public VoidObject VisitVarStmt(Stmt.Var stmt)
                 Resolve(stmt.Initializer);
             }
 
-            Define(stmt.Name, stmt.Initializer?.TypeReference ?? new TypeReference(stmt.Name));
+            Define(stmt.Name, stmt.TypeReference ?? new TypeReference(stmt.Name));
```

Comment:
This is an important part of this PR: the previous behavior was a bug; it would use the type of the _initializer_ even though that could potentially be of a different type than the actual variable being defined. (Think `var l: long = 123` and you get the picture; the variable should **not** become an `int`/`short` in this case)